air/src/execution_step/instructions/fail.rs

```diff
@@ -55,7 +55,7 @@ fn fail_with_scalar<'i>(scalar: &ast::Scalar<'i>, exec_ctx: &mut ExecutionCtx<'i
     let (value, mut tetraplet, provenance) = scalar.resolve(exec_ctx)?;
```

Comment:
I think we can call `check_error_object` only once in this file in the `fail_with_error_object` function.